[master/7.0] port version & upgrade matrix changes

wadells opened the pull request:
This is a forward port of the work in #1179 and #1195 to master/7.0.0.  The main purpose of this is to improve our PR & nightly build coverage scenarios.  This contributes substantially to https://github.com/gravitational/robotest/issues/155.

**Testing Done:**
```
walt@work:~/git/gravity$ make get-version
7.0.0-beta.1.4
```

I have not tried to run this robotest config in a try build (for either nightly or the PR build) because I didn't believe it an effective use of time, based on the level of testing I did for #1179 and #1195.  If we see failures or significant feedback I'll come back and dig into this testing.

wadells committed:
Bump "upgrade from" versions.

Use the latest 6.1.x and switch from 6.2.x (unsupported as of 2019-12-3)
to 6.3.x.

wadells committed:
Extend upgrade test matrix.

This patch gets us most of the value of:

  https://github.com/gravitational/robotest/issues/155

in our master (7.0.x) release branch with almost no effort.

(cherry picked from commit aa28377866bb56967c6a0d92e10488ce73f67299)

wadells committed:
Increment patch version for new content.

Before this commit, consecutive commits on top of tag x.y.z would result
in a version of x.y.z-c, where c was the number of commits since the
tag.  This was not semver compatible, because a hyphen is used to denote
a prerelease, as discussed at:

https://semver.org/#spec-item-9

Particularly: "Pre-release versions have a lower precedence than the
associated normal version."

After this patch, we increment the patch version before designating it
as a pre-release. This leads to semver compatible versioning like:

5.5.37 < 5.5.38-1 < 5.5.38

(cherry picked from commit 298a14e7d411ad80d717d186a20cfe92fc54c565)

wadells committed:
Add 'dev' signifier to post-release versions.

In order to help distinguish development builds, add 'dev' to the
pre-release portion of the semantic version string.

(cherry picked from commit ddadb12164be9ecc6aa5f2725644430c908be240)

a-palchikov commented:
There are a few upgrade test failures after the fact with degraded status - I'll create a separate issue for this.

wadells commented:
CI failures are all `6.3.0` - `this build` failures, where installing 6.3.0 on the node failed. The tests didn't get to the upgrade step.  `6.3.6` is fine, so I suspect this may be an issue that was already addressed in the branch.  Looking into the upgrade logs now.

wadells committed:
Disable upgrade testing from 6.3.0.

6.3.x prior to 341be36efe9c5e14f107b48122d43dd2b2910e4f is
non-determinstically healty (leaning towards never healthy based on my
testing) after install due to the following issue:

  https://github.com/gravitational/gravity/issues/1009

Thus 3.6.0 definitely isn't a good target for our CI.

I choose to leave 6.3.0 in the file with a comment to avoid others
trying it out and investing the same half day of learning & debugging
that I did.
